impl/src/main/java/io/kroxylicious/testing/kafka/common/KeystoreManager.java

```diff
@@ -0,0 +1,143 @@
+    }
+
+    /**
+     * Builds and adds provided certificate builder as a self-signed certificate
+     * @param certificateBuilder the builder configuring the certificate
+     * @return the self-signed certificate.
+     */
+    public X509Bundle createSelfSignedCertificate(CertificateBuilder certificateBuilder) throws Exception {
+        return certificateBuilder.copy()
+                .setIsCertificateAuthority(true)
+                .buildSelfSigned();
+    }
+
+    /**
+     * Optional we don't need this today!
+     * Builds and adds provided certificate builder as a certificate signed by the provided issuer.
+     * @param certificateBuilder the builder configuring the certificate
+     * @return the signed certificate.
+     */
+    public X509Bundle createSignedCertificate(X509Bundle issuer, CertificateBuilder certificateBuilder) throws Exception {
+        return certificateBuilder.copy()
+                .buildIssuedBy(issuer);
+    }
+
+    /**
+     * Formats the provided fields into a RFC5280 compliant form
+     * @param email the email
+     * @param domain the domain
+     * @param organizationUnit the organization unit
+     * @param organization the organization
+     * @param city the city
+     * @param state the state
+     * @param country the country
+     * @return the distinguished name
+     */
+    public String buildDistinguishedName(String email, String domain, String organizationUnit, String organization, String city, String state, String country) {
+        return "CN=" + domain + ", OU=" + organizationUnit + ", O=" + organization + ", L=" + city + ", ST=" + state + ", C=" + country + ", EMAILADDRESS=" + email;
+    }
+
+    /**
+     * Gets password.
+     *
+     * @return  the password
+     */
+    public String getPassword() {
+        if (this.password == null) {
+            this.password = UUID.randomUUID().toString().replace("-", "");
+        }
+
+        return this.password;
+    }
+
+    /**
+     * Generate certificate file path. It contains both certificates, subject's (key certs) and issuer's (trust CA certs)
+     * See {@link KeystoreManager#getPassword()} for getting the password
+     *
+     * @param bundle the bundle
+     * @return  the path of the generated certificate file
+     * @throws KeyStoreException the key store exception
+     * @throws IOException the io exception
+     * @throws CertificateException the certificate exception
+     * @throws NoSuchAlgorithmException the no such algorithm exception
+     */
+    public Path generateCertificateFile(X509Bundle bundle) throws KeyStoreException, IOException, CertificateException, NoSuchAlgorithmException {
+        KeyStore keyStore = bundle.toKeyStore(getPassword().toCharArray());
+        FileAttribute<Set<PosixFilePermission>> attr = PosixFilePermissions.asFileAttribute(PosixFilePermissions.fromString("rwx------"));
+        Path certsDirectory = Files.createTempDirectory("kroxylicious", attr);
```

Comment:
I'm tempted to make the certs directory a field and allow at least our unit tests to customise it, but I think it might be generally more useful to allow control over where the certs are generated.